test/e2e/azure_suite_test.go

```diff
@@ -138,11 +139,19 @@ var _ = BeforeSuite(func() {
 })
 
 var _ = AfterSuite(func() {
+	resourceName := mgmt.GetName()
 	// DO NOT stream "capi-controller-manager" logs as it prints out azure.json
 	Expect(writeLogs(mgmt, "capi-kubeadm-bootstrap-system", "capi-kubeadm-bootstrap-controller-manager", logPath)).To(Succeed())
 	Expect(writeLogs(mgmt, "capz-system", "capz-controller-manager", logPath)).To(Succeed())
 	By("Tearing down management cluster")
 	mgmt.Teardown(ctx)
+
+	By("Making sure there is no leftovers")
```

Comment:
We should do this in after each or collect all the names of the created resource groups and delete all of them as we might have created more than one (one per cluster but if we test several configurations like we do for single control plane and multi control plane there would be more than one).

Reply:
the AfterEach() where `ensureCAPZArtifactsDeleted` is called seems like a good place to do this

Reply:
I agree with moving this to `ensureCAPZArtifactsDeleted`. If the function does what it says, it seems it would include cleaning up any RG that was created.